From 278266fead9182f9b2f4fed1ac3b53f2c58c37d4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <cynerd@email.cz>
Date: Fri, 27 Aug 2021 13:00:56 +0200
Subject: [PATCH] base-files: Setup LEDs on Omnia

This configures network LEDs triggers for Turris Omnia on first boot.

It is not possible to simply autodetect PCIe cards and pair them with
correct slot as configuration is generated early during the first boot
when PCI devices are not yet fully initialized. Instead we set here only
good enough assumption of PCIe slots mapping to wireless interfaces.

Original author: Michal Hrusecky <Michal@Hrusecky.net>
---
 target/linux/mvebu/cortexa9/base-files/etc/board.d/01_leds | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/target/linux/mvebu/cortexa9/base-files/etc/board.d/01_leds b/target/linux/mvebu/cortexa9/base-files/etc/board.d/01_leds
index bfc589e6c0..f85af28f8f 100644
--- a/target/linux/mvebu/cortexa9/base-files/etc/board.d/01_leds
+++ b/target/linux/mvebu/cortexa9/base-files/etc/board.d/01_leds
@@ -15,6 +15,12 @@ ctera,c200-v2)
 	ucidef_set_led_usbport "usb2" "USB2" "green:usb-2" "usb1-port1" "usb2-port1"
 	ucidef_set_led_usbport "usb3" "USB3" "green:usb-1" "usb1-port2" "usb2-port2"
 	;;
+cznic,turris-omnia)
+	ucidef_set_led_netdev "wan" "wan" "omnia-led:wan" "eth2"
+	for i in $(seq 3); do
+		ucidef_set_led_netdev "pci$i" "pci$i" "omnia-led:pci$i" "wlan$i"
+	done
+	;;
 fortinet,fg-50e)
 	ucidef_set_led_netdev "wan1_link" "WAN1 Link" "green:speed_wan1" "eth1" "link"
 	ucidef_set_led_netdev "wan2_link" "WAN2 Link" "green:speed_wan2" "eth2" "link"
-- 
2.39.2

